File: innerlibs1/DebugTextWriter.vb
Imports System.IO
Imports System.Text

Public Class DebugTextWriter
    Inherits StreamWriter

    Public Sub New()
        MyBase.New(New DebugOutStream(), Encoding.Unicode, 1024)
        Me.AutoFlush = True
    End Sub



    NotInheritable Class DebugOutStream
        Inherits Stream

        Public Overrides Sub Write(ByVal buffer As Byte(), ByVal offset As Integer, ByVal count As Integer)
            Debug.Write(Encoding.Unicode.GetString(buffer, offset, count))
        End Sub

        Public Overrides ReadOnly Property CanRead As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property CanSeek As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property CanWrite As Boolean
            Get
                Return True
            End Get
        End Property



        Public Overrides ReadOnly Property Length As Long
            Get
                Throw bad_op
            End Get
        End Property

        Public Overrides Function Read(ByVal buffer As Byte(), ByVal offset As Integer, ByVal count As Integer) As Integer
            Throw bad_op
        End Function

        Public Overrides Function Seek(ByVal offset As Long, ByVal origin As SeekOrigin) As Long
            Throw bad_op
        End Function

        Public Overrides Sub SetLength(ByVal value As Long)
            Throw bad_op
        End Sub

        Public Overrides Sub Flush()
            Debug.Flush()
        End Sub

        Public Overrides Property Position As Long
            Get
                Throw bad_op
            End Get
            Set(ByVal value As Long)
                Throw bad_op
            End Set
        End Property

        Private Shared ReadOnly Property bad_op As InvalidOperationException
            Get
                Return New InvalidOperationException()
            End Get
        End Property


    End Class


End Class

File: InnerLibs/DebugTextWriter.cs
using System;
using System.Diagnostics;
using System.IO;
using System.Text;

namespace InnerLibs
{
    public class DebugTextWriter : StreamWriter
    {
        #region Public Constructors

        public DebugTextWriter() : base(new DebugOutStream(), Encoding.Unicode, 1024)
        {
            AutoFlush = true;
        }

        #endregion Public Constructors

      
    }
      public sealed class DebugOutStream : Stream
        {
            #region Private Properties

            private static InvalidOperationException Bad_op => new InvalidOperationException("Operation not supported");

            #endregion Private Properties

            #region Public Properties

            public override bool CanRead => false;

            public override bool CanSeek => false;

            public override bool CanWrite => true;

            public override long Length => throw Bad_op;

            public override long Position
            {
                get => throw Bad_op;

                set => throw Bad_op;
            }

            #endregion Public Properties

            #region Public Methods

            public override void Flush() => Debug.Flush();

            public override int Read(byte[] buffer, int offset, int count) => throw Bad_op;

            public override long Seek(long offset, SeekOrigin origin) => throw Bad_op;

            public override void SetLength(long value) => throw Bad_op;

            public override void Write(byte[] buffer, int offset, int count) => Debug.Write(Encoding.Unicode.GetString(buffer, offset, count));

            #endregion Public Methods
        }

}